File: schemas/cms-blog/drizzle/comment_votes.ts
// comment_votes: User upvotes and downvotes on comments.
// See README.md for full design rationale.
import { pgTable, uuid, integer, timestamp, unique, index } from "drizzle-orm/pg-core";
import { comments } from "./comments";

export const commentVotes = pgTable("comment_votes", {
  id: uuid("id").primaryKey().defaultRandom(),
  commentId: uuid("comment_id").notNull().references(() => comments.id, { onDelete: "cascade" }),
  userId: uuid("user_id").notNull(), // FK → users.id (cascade delete)
  value: integer("value").notNull(),
  createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
}, (table) => [
  unique("uq_comment_votes_comment_id_user_id").on(table.commentId, table.userId),
  index("idx_comment_votes_user_id").on(table.userId),
]);

File: schemas/cms-blog/spacetimedb/comment_votes.rs
// comment_votes: User votes on comments for community ranking.
// See README.md for full design rationale.
use spacetimedb::Timestamp;

#[spacetimedb::table(name = comment_votes, public)]
pub struct CommentVote {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub comment_id: String, // FK -> comments.id (cascade delete)
    #[index(btree)]
    pub user_id: String, // FK -> users.id (cascade delete)
    pub value: i32, // Composite unique: (comment_id, user_id)
    pub created_at: Timestamp,
}
